src/System.IO.Pipes/tests/PipeTest.Read.cs

```diff
@@ -17,10 +17,15 @@ namespace System.IO.Pipes.Tests
     public abstract partial class PipeTest_Read : PipeTestBase
     {
         public virtual bool SupportsBidirectionalReadingWriting => false;
+        protected virtual bool ShouldSkipTest() => false;
 
         [Fact]
         public void ReadWithNullBuffer_Throws_ArgumentNullException()
         {
+            if (ShouldSkipTest())
```

Comment:
>ShouldSkipTest() [](start = 16, length = 16)

I guess this wouldn't work in a ConditionalFact attribute. If it did it will report correctly that the test was skipped.

Reply:
it won't, the problem here is that this is an abstract test class, so only subclasses run the the tests, but we want to run them conditionally. If you check the original PR(the one against release/uwp6.0 branch) you can find more info on why did we have to do it this way instead.